### ISSUE TITLE
SteamOS and GTA 5, input/focus issues

### ISSUE THREAD

**Issue by steamoshza** [ValveSoftware/Proton#1170]
hello at valve!

First of all: thank you so much for your efforts, steam play seams to open totally new possibilities, at least for legacy games!
Also I have tested proton with another game, some half life 2 mod, it runs really very much perfectly fine with steamOS.

GTA 5 however, causes issues - I'd like to help you to the best of my abilities to resolve this, since I really would love to have it.

The issue starts with the fact that I am not using any desktop distro for this, but SteamOS. The latest steamos-beta is installed (mostly for the nvidia drivers and probably mesa), the game detects and runs with DirectX 11, the performance seems to be very good.
I had to install msttcorefonts and manually copy over the fonts via

cp /usr/share/fonts/truetype/msttcorefonts/* /home/steam/.local/share/Steam/steamapps/compatdata/271590/pfx/drive_c/windows/system32/fonts/

.. since you do not provide winetricks to install them and trying to install winetricks from different sources caused dependency problems. without those fonts the launcher simply did not start as it should.

This way I could start it, however, the launcher did not accept any input from the onscreen keyboard. I could resolve this by streaming the game to my PC, where I was able to use the keyboard and CRTL+V the provided CD-key.
After that, the game did start, but stutters frequently, although the FPS appear to be stable when in non-stutter and decent, with all settings maxed out.

The show stopper is this: the steam controller flashes all the time and every input seems to map to the controller AND mouse-keyboard *at once*.
NOTE: please DO NOT CONFUSE THIS with the usual 'games switch automatically between mouse+keyboard and gamepad, whatever is being used'; I am not touching anything, and the switching occurs all the time, like 10 times per second.
Usually the controller LED should be white, but it keeps flashing like 100/60/100/60 percent all the time - I set the brightness to ~60% as my preference, so this is how I can observe that flashing which typically indicates that the controller is (re-)connecting.
However, there is no overlay info in the bottom right corner telling me that the controller is reconnecting, it's just flashing.

This has to be some focus issue with big picture, because once I press the steam home button, the steam overlay *also* does not respond! Effectively no buttons are working then, I literally have to restart the system or kill -9 GTA5.exe via SSH to regain control of the host.
There is absolutely no input accepted from this point on.

Anyway, all of this is unplayable because of this, and I do suspect that (especially because the steam overlay fails to respond) this is more or less some focus issue, which you would be able to fix, hopefully.

In short, **how to reproduce this**: just install GTA 5 on steamOS, you may want to use my workaround to input your CD key and start the game, ...and then immediately select STORY MODE (it will continue with ONLINE MODE otherwise), and you should be able to see for yourselves.

Please tell me if you have the time or resources to dedicate for this issue – I am a dev myself and do see that this may be something not even worth looking into, but keep in mind that there are many people still playing GTA, and at least that focus issue seems fixable to me, regardless what the game seems to do.

Also, before starting I had to apply this: https://github.com/ValveSoftware/Proton/issues/37#issuecomment-415833819

Thank you Valve!

**Comment by kisak-valve** [comment]
Hello @steamoshza, we'll be using one issue report per unofficially supported game title, so I've gone ahead and transferred this issue report to https://github.com/ValveSoftware/Proton/issues/37#issuecomment-417925826.

**Comment by steamoshza** [comment]
well I co not concur with that, since I do not believe that this has specifically to do something with the game.
regardless what it does, totally losing control/focus of steam, the overlay or any input can't be that much related to the game. it is merely an example for which game this happens with.

so my best guess is that this has to do directly with proton or BPM, GTA must not be able to cause such a severely unresponsive system.
